File: AutomaticDrawing.Recognition.FastAI/Recognition/FastRecognition.vb
Imports AutoPaint.Core
Imports AutoPaint.Utilities
''' <summary>
''' 使用快速模型的线条识别器
''' </summary>
Public Class FastRecognition
    Implements IRecognition
    Public Function Recognize(pixels As PixelData) As List(Of ILine) Implements IRecognition.Recognize
        Dim sequenceAI As New SequenceAI(ColorHelper.GetPixelDataBools(pixels))
        Return sequenceAI.Lines
    End Function
End Class

File: AutoPaint.Recognition.FastAI/Recognition/FastRecognition.cs
using AutoPaint.Core;
using AutoPaint.Utilities;
using System;
using System.Collections.Generic;
using System.Text;

namespace AutoPaint.Recognition
{
    public class FastRecognition : IRecognition
    {
        public List<ILine> Recognize(PixelData pixels)
        {
            FastAI sequenceAI = new FastAI(ColorHelper.GetPixelDataBools(pixels));
            return sequenceAI.Lines;
        }

        IPainting IRecognition.Recognize(PixelData pixels)
        {
            FastAI sequenceAI = new FastAI(ColorHelper.GetPixelDataBools(pixels));
            return new Painting() { Lines = sequenceAI.Lines };
        }
    }
}
